File: nim/cksum.nim
import strutils
import sequtils

#for part 1
proc diff(row: seq[int]): int =
  return (max(row) - min(row))

#for part2
proc modd(row: seq[int]): int =
  for i,v1 in row:
    for j,v2 in row:
      if v1 == v2:
        continue
      if v1 mod v2 == 0:
        return int(v1/v2)
  

proc cksum(f: string): int =
  var sum = 0
  for line in lines f:
    sum += modd(map(line.split, parseInt))
  return sum

echo cksum("cksumin.txt")

File: python/cksum.py
#!/usr/local/bin/python3

#for part 1
def diff(row):
    return abs(max(row) - min(row))

#for part 2
def mod(row):
    for i in row:
        for j in row:
            if i == j:
                continue
            if i % j == 0:
                return i/j
            
def cksum(filename):
    f = open(filename, 'r')
    cksum = 0
    cksum = [mod([int(x) for x in line.split()]) for line in f]
    return sum(cksum)

print(cksum("cksumin.txt"))
